### PR TITLE
Generate GAPIC Data Types docs TOCs

### LINKED ISSUE #1180: Generate content automatically for the table shown on the 'Data Types' page.
Right now, the content is manually added to the generated dummy ruby classes. We need to either generate the content for the table or add Angular app logic to extract the data from the ruby classes in order to display the table.

### LINKED ISSUE #1206: Document Vision low-level API (GAPIC code)
Types such as [v1/image_annotator_client.rb](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-vision/lib/google/cloud/vision/v1/image_annotator_client.rb) are not currently published to the gh-pages documentation.

- [] Remove exclusions from `.yardopts`
- [] Add navigation to GAPIC/gRPC code

### PR DESCRIPTION
**Pull request by quartzmo**
Replace hand-written "Data Types" TOCs (as specified in GoogleCloudPlatform/gcloud-common/issues/207) in GAPIC code with generated jsondoc files.

Add GAPIC docs to Vision.

[closes #1180, closes #1206]

- [x] Datastore
- [x] Error Reporting
- [x] Language
- [x] Logging
- [x] Monitoring 
- [x] Pub/Sub
- [x] Spanner
- [x] Trace
- [x] Vision

### CODE REVIEW

**Comment by quartzmo** [comment]
@blowmage I can screenshare a demo with you.

File: google-cloud-spanner/docs/toc.json [hunk 1]
```diff
@@ -31,7 +31,25 @@
...
+      "type": "google/cloud/spanner",
+      "patterns": ["\/spanner", "google\/protobuf"],
+      "nav": [
+        {
+          "title": "V1",
+          "type": "google/cloud/spanner/v1",
+          "patterns": ["\/spanner\/v1", "google\/protobuf"],
+          "nav": [
+            {
+              "title": "SpannerClient",
```
Comment (blowmage) [inline google-cloud-spanner/docs/toc.json]: Admin/Instances/V1/InstanceClient and Admin/Databases/V1/DatabaseClient should also be listed here.

File: google-cloud-spanner/Rakefile [hunk 2]
```diff
@@ -128,7 +128,29 @@ task :jsondoc => :yard do
...
-  generator = Gcloud::Jsondoc::Generator.new registry, "google-cloud-spanner"
+
+  toc_config = {
+    documents: [
+      {
+        type: "toc",
+        title: "Google::Spanner::V1::DataTypes",
+        modules: [
+          {
+            title: "Google::Spanner::V1",
```
Comment (blowmage) [inline google-cloud-spanner/Rakefile]: admin/instances/v1 and admin/databases/v1 should be included here as well.

File: google-cloud-spanner/docs/toc.json (hunk 1 above)
Comment (quartzmo) [inline google-cloud-spanner/docs/toc.json]: I don't see these files in `lib/google/cloud/spanner/v1/doc`, do they still need to be added?

Comment (blowmage) [inline google-cloud-spanner/docs/toc.json]: Yep.

**Pull request merged**